examples/options/src/clusterization.cpp

```diff
@@ -15,20 +15,43 @@ namespace traccc::opts {
 
 clusterization::clusterization() : interface("Clusterization Options") {
 
+    m_desc.add_options()("threads-per-partition",
+                         boost::program_options::value(&threads_per_partition)
+                             ->default_value(256),
+                         "The number of threads per partition");
     m_desc.add_options()(
-        "target-cells-per-partition",
-        boost::program_options::value(&target_cells_per_partition)
-            ->default_value(target_cells_per_partition),
-        "The number of cells to merge in a partition");
+        "max-cells-per-thread",
+        boost::program_options::value(&max_cells_per_thread)->default_value(16),
+        "The maximum number of cells per thread");
+    m_desc.add_options()("target-cells-per-thread",
+                         boost::program_options::value(&target_cells_per_thread)
+                             ->default_value(8),
+                         "The target number of cells per thread");
+    m_desc.add_options()("backup-size-multiplier",
+                         boost::program_options::value(&backup_size_multiplier)
+                             ->default_value(256),
+                         "The size multiplier of the backup scratch space");
 }
 
-clusterization::operator unsigned short() const {
-    return target_cells_per_partition;
+clusterization::operator clustering_config() const {
+    clustering_config rv;
+
+    rv.threads_per_partition = threads_per_partition;
+    rv.max_cells_per_thread = max_cells_per_thread;
+    rv.target_cells_per_thread = target_cells_per_thread;
+    rv.backup_size_multiplier = backup_size_multiplier;
+
+    return rv;
 }
 
 std::ostream& clusterization::print_impl(std::ostream& out) const {
-
-    out << "  Target cells per partition: " << target_cells_per_partition;
+    out << "  Threads per partition:      " << threads_per_partition
+        << std::endl;
+    out << "  Target cells per thread:    " << target_cells_per_thread
+        << std::endl;
+    out << "  Max cells per thread:       " << max_cells_per_thread
+        << std::endl;
```

Comment:
I'm a bit surprised that you went with 3 `std::endl` calls here. :confused: I prefer just using "`\n`" in such places. The user can decide when they want to flush the stream, themselves.